src/buf_encoder.rs

```diff
@@ -19,6 +19,22 @@ use super::{Case, Table};
 /// This is an important building block for fast hex-encoding. Because string writing tools
 /// provided by `core::fmt` involve dynamic dispatch and don't allow reserving capacity in strings
 /// buffering the hex and then formatting it is significantly faster.
+///
+/// # Examples
+/// ```
+/// # use hex_conservative::buf_encoder::BufEncoder;
+/// # use hex_conservative::Case;
+/// let mut encoder = BufEncoder::<4>::new(Case::Lower);
+/// encoder.put_byte(0xab);
+/// assert_eq!(encoder.as_str(), "ab");
+/// ```
+/// ```compile_fail
```

Comment:
Documenting it is a good idea. But I'd place some explanation before the block start.

Reply:
I added an explanation that the capacity has to be even, which does now make the the examples fit in better.

Reply:
I meant before the failing block. Something like "The following code doesn't compile because of odd capacity".

Reply:
Done